test/integration/element_tracing.cc

```diff
@@ -0,0 +1,231 @@
+
+  const std::string childLinkXmlPath = xmlPath + "/link[@name=\"child\"]";
+
+  const sdf::Link *childLink = model->LinkByName("child");
+  ASSERT_NE(nullptr, childLink);
+  sdf::ElementPtr childLinkElem = childLink->Element();
+  checkElementPtr(childLinkElem, testFile, 4, childLinkXmlPath);
+
+  const std::string jointXmlPath =
+    xmlPath + "/joint[@name=\"top_level_joint\"]";
+
+  const sdf::Joint *joint = model->JointByIndex(0);
+  ASSERT_NE(nullptr, joint);
+  sdf::ElementPtr jointElem = joint->Element();
+  checkElementPtr(jointElem, testFile, 11, jointXmlPath);
+
+  const std::string nestedModelXmlPath =
+    xmlPath + "/model[@name=\"nested_model\"]";
+
+  const sdf::Model *nestedModel = model->ModelByName("nested_model");
+  ASSERT_NE(nullptr, nestedModel);
+  sdf::ElementPtr nestedModelElem = nestedModel->Element();
+  checkElementPtr(nestedModelElem, testFile, 5, nestedModelXmlPath);
+
+  const std::string nestedLinkXmlPath =
+    nestedModelXmlPath + "/link[@name=\"nested_link01\"]";
+
+  const sdf::Link *nestedLink = nestedModel->LinkByName("nested_link01");
+  ASSERT_NE(nullptr, nestedLink);
+  sdf::ElementPtr nestedLinkElem = nestedLink->Element();
+  checkElementPtr(nestedLinkElem, testFile, 6, nestedLinkXmlPath);
+
+  const std::string nestedNestedModelXmlPath =
+    nestedModelXmlPath + "/model[@name=\"nested_nested_model\"]";
+
+  const sdf::Model *nestedNestedModel =
+    nestedModel->ModelByName("nested_nested_model");
+  ASSERT_NE(nullptr, nestedNestedModel);
+  sdf::ElementPtr nestedNestedModelElem = nestedNestedModel->Element();
+  checkElementPtr(nestedNestedModelElem, testFile, 7,
+      nestedNestedModelXmlPath);
+
+  const std::string nestedNestedLinkXmlPath =
+    nestedNestedModelXmlPath + "/link[@name=\"nested_nested_link01\"]";
+
+  const sdf::Link *nestedNestedLink =
+    nestedNestedModel->LinkByName("nested_nested_link01");
+  ASSERT_NE(nullptr, nestedNestedLink);
+  sdf::ElementPtr nestedNestedLinkElem = nestedNestedLink->Element();
+  checkElementPtr(nestedNestedLinkElem, testFile, 8, nestedNestedLinkXmlPath);
+}
+
+//////////////////////////////////////////////////
+TEST(ElementTracing, includes)
+{
+  sdf::setFindCallback(findFileCb);
+
+  const auto worldFile = sdf::testing::TestFile("sdf", "includes.sdf");
+  sdf::Root root;
+  sdf::Errors errors = root.Load(worldFile);
+  EXPECT_TRUE(errors.empty());
+
+  const std::string xmlPath = "/sdf";
+
+  sdf::ElementPtr rootElem = root.Element();
+  checkElementPtr(rootElem, worldFile, 2, xmlPath);
+
+  const std::string worldXmlPath = xmlPath + "/world[@name=\"default\"]";
+  const sdf::World *world = root.WorldByIndex(0);
+  ASSERT_NE(nullptr, world);
+  sdf::ElementPtr worldElem = world->Element();
+  checkElementPtr(worldElem, worldFile, 3, worldXmlPath);
+
+  // Actors
+  const std::string actorFilePath =
+      sdf::testing::TestFile("integration", "model", "test_actor") +
+      "/model.sdf";
```

Comment:
I believe windows CI is failing because of these `+ "/model.sdf"`, which should be fixed with adding `"model.sdf"` as a parameter:
```suggestion
  const std::string actorFilePath =
      sdf::testing::TestFile("integration", "model", "test_actor", "model.sdf");
```

Reply:
Thanks! I didn't know how to use `testing::TestFile` properly! https://github.com/osrf/sdformat/pull/548/commits/ee7f583a830334d39adfc9a2443fab96ac433b11